source/server/connection_handler_impl.cc

```diff
@@ -52,133 +56,142 @@ void ConnectionHandlerImpl::addListener(absl::optional<uint64_t> overridden_list
     if (overridden_listener.has_value()) {
       if (auto iter = listener_map_by_tag_.find(overridden_listener.value());
           iter != listener_map_by_tag_.end()) {
-        iter->second->internalListener()->updateListenerConfig(config);
+        iter->second->invokeListenerMethod(
+            [&config](Network::ConnectionHandler::ActiveListener& listener) {
+              listener.updateListenerConfig(config);
+            });
         return;
       }
       IS_ENVOY_BUG("unexpected");
     }
     auto internal_listener =
         local_registry->createActiveInternalListener(*this, config, dispatcher());
-    details->typed_listener_ = *internal_listener;
-    details->listener_ = std::move(internal_listener);
-  } else if (config.listenSocketFactory().socketType() == Network::Socket::Type::Stream) {
+    // TODO(soulxu): support multiple internal addresses in listener in the future.
+    ASSERT(config.listenSocketFactories().size() == 1);
```

Comment:
Does this assert true also before the enclosing `if` scope?
Seeing the `else if` statement on line 73 suggests that this is true, but not sure.

Reply:
This is for the multiple addresses doesn't support internal address. In the late PR, the API will reject multiple internal addresses, so I have the assertion here.